rp2040-hal/src/pio.rs

```diff
@@ -500,56 +522,58 @@ impl<SM: ValidStateMachine> StateMachine<SM, Stopped> {
         }
     }
 
-    /// Sets the pin directions for the specified pins.
+    /// Sets the pin state for the specified pins.
     ///
-    /// The `pins` parameter specifies a set of pins as a mask, and `pindir` contains the
-    /// directions that are configured for these pins. The bits in both masks correspond to the pin
-    /// number. The user has to make sure that they do not select any pins that are in use by any
+    /// The user has to make sure that they do not select any pins that are in use by any
     /// other state machines of the same PIO block.
     ///
-    /// This function needs to be called for sideset pins if they are supposed to be used as
-    /// output pins.
-    pub fn set_pindirs_with_mask(&mut self, mut pins: u32, pindir: u32) {
-        let mut pin = 0;
-        let prev_pinctrl = self.sm.sm().sm_pinctrl.read().bits();
-        // For each pin in the mask, we select the pin as a SET pin and then execute "set PINDIRS,
-        // <direction>".
-        while pins != 0 {
-            if (pins & 1) != 0 {
-                self.sm.sm().sm_pinctrl.write(|w| {
-                    unsafe {
-                        w.set_count().bits(1);
-                        w.set_base().bits(pin as u8);
-                    }
-                    w
-                });
-                let set_pindirs = pio::Instruction {
-                    operands: pio::InstructionOperands::SET {
-                        destination: pio::SetDestination::PINDIRS,
-                        data: ((pindir >> pin) & 0x1) as u8,
-                    },
-                    delay: 0,
-                    side_set: None,
+    /// The iterator's item are pairs of `(pin_number, pin_state)`.
+    pub fn set_pins_with_iter(&mut self, pins: impl IntoIterator<Item = (u8, PinState)>) {
```

Comment:
I wondered why the method includes the suffix `_with_iter` - perhaps to distinguish it from `PIOBuilder::set_pins`?
In that case I'd expect that both methods do basically the same, just with different parameters. Which is not the case, they do fundamentally different things.
If it is because you expect that there may be different variants of this method in the future, the suffix is fine.
(I'd prefer to have a stronger distinction from `PIOBuilder::set_pins`, but have no idea for a better name, either.)